atomate/cp2k/firetasks/glue_tasks.py

```diff
@@ -0,0 +1,147 @@
+# coding: utf-8
+
+from __future__ import division, print_function, unicode_literals, \
+    absolute_import
+
+import glob
+
+from pymatgen.analysis.elasticity.strain import Strain
+from pymatgen.io.vasp import Vasprun, zpath
+
+"""
+This module defines tasks that acts as a glue between other vasp Firetasks to allow communication
+between different Firetasks and Fireworks. This module also contains tasks that affect the control
+flow of the workflow, e.g. tasks to check stability or the gap is within a certain range.
+"""
+
+import gzip
+import os
+import re
+import pickle
+import glob
+from monty.io import zopen
+
+from pymatgen.io.cp2k.sets import Cp2kInputSet
+from pymatgen.io.cp2k.inputs import Cp2kInput, Coord, Cell
+from pymatgen.io.cp2k.outputs import Cp2kOutput
+
+from fireworks import explicit_serialize, FiretaskBase, FWAction
+
+from atomate.utils.utils import env_chk, get_logger
+from atomate.common.firetasks.glue_tasks import get_calc_loc, PassResult, \
+    CopyFiles, CopyFilesFromCalcLoc
+
+logger = get_logger(__name__)
+
+__author__ = 'Nicholas Winner'
+__email__ = 'nwinner@berkeley.edu'
+
+
+@explicit_serialize
+class UpdateStructureFromPrevCalc(FiretaskBase):
+    """
+    Using the location of a previous calculation. The CP2K output parser will
+    get the final structure from a previous calculation and update this FW's
```

Comment:
Is there a reason why you are updating the structure by updating the firework spec rather than either:
1. Just writing the structure to file
2. Doing the structure updating as part of `WriteCp2kFromPrevious`.

I think option 2 makes the most sense. If you only want to copy the structure and nothing else from the previous calculation, you could add options to that effect in `WriteCp2kFromPrevious`.

Reply:
Hey Alex, number 2 probably does make more sense. Number 1 isn't the case because cp2k doesn't use the same formatting as VASP where a separate file holds the structure. Generally cp2k has everything in a single input file. This can actually be modified, but its not a cp2k handles stuff by default so i didn't want to complicate that until a later time.

Reply:
Ok great. Let's go with option 2!